src/Libraries/Microsoft.Extensions.AI.Evaluation.Reporting/CSharp/ScenarioRunResult.cs

```diff
@@ -81,6 +83,11 @@ public ScenarioRunResult(
     {
     }
 
+    /// <summary>
+    /// Gets or sets the version of the format used to persist the current <see cref="ScenarioRunResult"/>.
+    /// </summary>
+    public int? FormatVersion { get; set; } = formatVersion ?? Defaults.ReportingFormatVersion;
```

Comment:
I suggest this should instead be:
```suggestion
    public int? FormatVersion { get; set; } = formatVersion;
```
or:
```suggestion
public int FormatVersion { get; set; } = formatVersion ?? Defaults.ReportingFormatVersion;
```

Reply:
I was under the impression that the type in the JSON constructor must be the same as the property type for the serialization logic to work. But I could be wrong about that. I'll test it.

Reply:
Types need to match with the parameter.

Reply:
> I was under the impression that the type in the JSON constructor must be the same as the property type for the serialization logic to work

Yes, confirmed that if the constructor is `int?` then the property type must match it. So the second option won't work.

I don't understand the reason for the first suggestion. We want to have the version set to the default in the library, so we don't have to remember to set it every time we use `ScenarioRunResult`. The only reason the property is settable, is because it is a JSON serializable object. I suppose we could make it readonly.

Reply:
> Types need to match with the parameter.

Only if the parameter is used for roundtripping, right? There's no mechanism to say "yes, I see there's a property with the same name as this parameter, but the types don't match and that property's setter should be used instead of the constructor"?

Reply:
If you're happy with it as is it, that's fine, don't let my feedback stop you.

It's just weird to me that the property is nullable but defaults to a non-null value, can then be set to null, but really shouldn't be, and if it is set to null, then it behaves differently than if null is provided to the constructor, as if null is provided to the constructor, it'll then default to the default value.

Reply:
> Only if the parameter is used for roundtripping, right? There's no mechanism to say "yes, I see there's a property with the same name as this parameter, but the types don't match and that property's setter should be used instead of the constructor"?

I think there is such a mechanism. When I try it the Deserialize method throws this exception...

> System.InvalidOperationException : Each parameter in the deserialization constructor on type 'Microsoft.Extensions.AI.Evaluation.Reporting.ScenarioRunResult' must bind to an object property or field on deserialization. Each parameter name must match with a property or field on the object. Fields are only considered when 'JsonSerializerOptions.IncludeFields' is enabled. The match can be case-insensitive.

What if we make the property readonly, but settable through the json constructor? I believe that works as expected.

Reply:
> If you're happy with it as is it, that's fine, don't let my feedback stop you.

I do still need an approval from @dotnet/dotnet-extensions-ai, due to the change in the CacheHelper.

Reply:
> > Types need to match with the parameter.
> 
> Only if the parameter is used for roundtripping, right? There's no mechanism to say "yes, I see there's a property with the same name as this parameter, but the types don't match and that property's setter should be used instead of the constructor"?

There's no such mechanism present today, unfortunately. We've received a lot of feedback about this kind of enforcement being needlessly strict but we just haven't gotten around to relaxing it or offering any degree of configurability.